tck/src/main/java/org/eclipse/microprofile/concurrency/tck/MPConfigTest.java

```diff
@@ -405,6 +409,58 @@ public void overrideContextPropagationForThreadContextParameterWithConfig() {
             Thread.currentThread().setPriority(originalPriority);
         }
     }
+    
+    /**
+     * Verify MicroProfile Config overrides the propagated attribute to enabled
+     * an implied clear of all remaining.
+     * @throws Exception 
+     */
+    @Test
+    public void overrideContextPropagationForThreadContextWithImpliedCleared() throws Exception {
+        // Expected config is propagated=Buffer; cleared=Transaction(with implied Remaining); unchanged=Label
+        Assert.assertNotNull(clearAllRemainingThreadContext,
+                "Unable to inject ThreadContext qualified by NamedInstance. Cannot run test.");
+        
+        int originalPriority = Thread.currentThread().getPriority();     
+        try {
+            // Set non-default values
+            int newPriority = originalPriority == 3 ? 2 : 3;
+            Thread.currentThread().setPriority(newPriority);
+            Buffer.set(new StringBuffer("clearUnspecifiedContexts-test-buffer-A"));
+            Label.set("clearUnspecifiedContexts-test-label-A");
+
+            Callable<Integer> callable = clearAllRemainingThreadContext.contextualCallable(() -> {
+                    Assert.assertEquals(Buffer.get().toString(), "clearUnspecifiedContexts-test-buffer-A",
+                            "Context type was not propagated to contextual action.");
+
+                    Assert.assertEquals(Label.get(), "clearUnspecifiedContexts-test-label-C",
+                            "Context type was not left unchanged by contextual action.");
+
+                    Buffer.set(new StringBuffer("clearUnspecifiedContexts-test-buffer-B"));
+                    Label.set("clearUnspecifiedContexts-test-label-B");
+
+                    return Thread.currentThread().getPriority();
+            });
+
+            Buffer.set(new StringBuffer("clearUnspecifiedContexts-test-buffer-C"));
+            Label.set("clearUnspecifiedContexts-test-label-C");
+
+            Assert.assertEquals(callable.call(), Integer.valueOf(Thread.NORM_PRIORITY),
+                    "Context type that remained unspecified was not cleared by default.");
+            
+            Assert.assertEquals(Buffer.get().toString(), "clearUnspecifiedContexts-test-buffer-C",
+                    "Context type was not left unchanged by contextual action.");
```

Comment:
This context type isn't left unchanged. It is propagated and then restored afterward. So the test here is correct, but the messages is wrong.  Should be "Context type was not restored after contextual action."